added documentation for retrieving dump files (#3586)
dotnet/arcade: Documentation/Dumps/Dumps.md

# Retrieving Dump Files and Other Logs From a Crash or Timeout

> If you encounter a crash or a timeout during a test run, you can still access any files that were uploaded during the lifetime of the workitem that encountered the issue. 

Reviewer: I think there should be some context around this procedure to connect it to Helix's general behavior. This is framed for crash dumps specifically (which is very important) but the procedure is useful for other failure information.

```suggestion
Crash dumps are uploaded to an Azure storage account in the same manner as test results, logs, and other output related to the work item. When a work item contains a failure, a special file containing links to these files is created and attached to a test result named "{Test} Failure Results". It may be viewed along with the other tests in the Azure DevOps Test tab.
```

Here are steps to find the dump files: 

1. Navigate to the Test tab in Azure DevOps: 

![AzDO Test Tab](DumpsTestTab.png)

> 2. Click the Outcome dropdown (circled in blue), and click "X Clear" or select "Others". 

Reviewer: ```suggestion
2. By default, only test results without outcome "aborted" or "failed" are shown. The Failure Results test result is marked with an outcome of "None." To filter for this, click the Outcome dropdown (circled in blue), and click "X Clear" or select "Others". 
```

> The Test Result that contains the link to the dumps and other files is of type "Error" which shows up under "Others". 

Reviewer 2: ```suggestion
The test result that contains the link to the dumps and other files is of type "Error" which shows up under "Others."
```

![Outcome dropdown](DumpsOutcomeDropdown.png)

3. Filter by "Failure Report" in the "Filter by test or run name" text box (circled in red). 

> The Test Result will be named: workitem name + "Failure Report"

Reviewer 2: ```suggestion
The test result will be named `[workitem_name] Failure Report`.
```
I'm guessing on what the file name is here, but it's a bit unclear to me how it will be named.

> 2. By default, only test results without outcome "aborted" or "failed" are shown. The Failure Results test result is marked with an outcome of "None." To filter for this, click the Outcome dropdown (circled in blue), and click "X Clear" or select "Others". 

Reviewer: ```suggestion
2. By default, only test results with outcome "aborted" or "failed" are shown. The Failure Results test result is marked with an outcome of "None." To filter for this, click the Outcome dropdown (circled in blue), and click "X Clear" or select "Others". 
```

![Failure Report Filter](DumpsFailureReportFilter.png)

> 4. After filtering down to the Failure Reports, you should see the Test Results of the appropriate name. You may need to expand the Test Run to see the Failure Report Test Result. 

Reviewer 2: ```suggestion
4. After filtering down to the Failure Reports, you should see the test results matching the provided name. You may need to expand the Test Run to see the Failure Report Test Result. 
```
add descriptions for what we're expanding and seeing (e.g. "Test Run dropdown accordion").

![Failure Report Test Results](DumpsFailureReportTestResults.png)

> 5. Click on the Test Result link to open up the Test Result blade. Click on the Attachments tab to view the attached UploadFileResult.html file. 

Reviewer 2: ```suggestion
5. Click on the Test Result link to open up the Test Result blade. Click on the Attachments tab to view the attached `UploadFileResult.html` file. 
```
